Part1_InterruptControllersEvolution.html

```diff
@@ -104,79 +104,78 @@ <h3>PIC</h3>
 		<td class="tg-0lax">Data</td>
 		<td class="tg-0lax">0x00A1</td>
 	  </tr></table>
-	<p>Full documentation of the 8259A can be found <a href="https://pdos.csail.mit.edu/6.828/2005/readings/hardware/8259A.pdf">here</a><p/>
+	<p>Full documentation of the 8259A can be found <a href="https://pdos.csail.mit.edu/6.828/2005/readings/hardware/8259A.pdf">here</a><p/>.
 
-	<p>The PCI bus has replaced the ISA bus. And the number of devices amount clearly began to 
-	exceed the number 15. Plus instead of the static ISA bus in the PCI bus devices can be added to system dynamically. But luckily, in the PCI bus interrupts can be shared (it means that it is possible to connect many devices to one interrupt line IRQ). In the end, to solve the problem of lack of interrupt lines, it was decided to group interrupts from all of PCI devices to PIRQ lines (Programmable Interrupt Request).<p/>
+	<p>The PCI bus later replaced the ISA bus, and the number of devices began to exceed the number 15. Instead of the static ISA bus, devices in the PCI bus can be added to the system dynamically. In the PCI bus, interrupts can also be shared, since it is possible to connect many devices to one interrupt line IRQ. In the end, to solve the problem of lack of interrupt lines, it was decided to group interrupts from all of the PCI devices to PIRQ lines (Programmable Interrupt Request).<p/>

```

Comment:
I really meant it like this:
downside: "and the number of devices began to exceed the number 15"
downside: "and instead of the static ISA bus, devices in the PCI bus can be added to the system dynamically"
So all is bad, but...
"But luckily, in the PCI bus interrupts can be shared"

So all 3 sentences were connected with one idea. I couldn't read it in the new variant. It is like all these 3 sentences are independent from each other.

Reply:
Ah ok, I didn't see that those were downsides. We could say something like "Unfortunately, the number of devices began to exceed the number 15, and instead of the static ISA bus, devices in the PCI bus can be added to the system dynamically. But luckily, interrupts in the PCI bus can be shared, since ..."

Reply:
What about this variant?

```suggestion
	<p>The PCI bus later replaced the ISA bus. Unfortunately, the number of devices began to exceed the number 15. Also instead of the static ISA bus, devices in the PCI bus can be added to the system dynamically which could potentially lead to even more problems. But luckily, interrupts in the PCI bus can be shared, therefore it is possible to connect many devices to one interrupt line IRQ. In the end, to solve the problem of lack of interrupt lines, it was decided to group interrupts from all of the PCI devices to PIRQ lines (Programmable Interrupt Request).<p/>
```

Reply:
I would use "so" instead of "therefore". "therefore" is mostly used if you're making a conclusion at the end of a big logical argument. "so" is better for a simple implication. Otherwise it sounds good.

Reply:
```suggestion
	<p>The PCI bus later replaced the ISA bus. Unfortunately, the number of devices began to exceed the number 15. Also instead of the static ISA bus, devices in the PCI bus can be added to the system dynamically which could potentially lead to even more problems. But luckily, interrupts in the PCI bus can be shared, so it is possible to connect many devices to one interrupt line IRQ. In the end, to solve the problem of lack of interrupt lines, it was decided to group interrupts from all of the PCI devices to PIRQ lines (Programmable Interrupt Request).<p/>
```